sycl/source/detail/platform_impl.hpp

```diff
@@ -120,6 +120,14 @@ class platform_impl {
     return *MPlugin;
   }
 
+  /// Sets the platform implementation to use another plugin.
+  ///
+  /// \param PluginPtr is a pointer to a plugin instance
+  void setPlugin(std::shared_ptr<plugin> PluginPtr) {
+    assert(!MHostPlatform && "Plugin is not available for Host");
```

Comment:
FYI you may not execute this assert if `MPlatform` is `nullptr` in context or elsewhere it is stored in.

Reply:
TBH, I don't see how this would affect an assertion that checks a `bool` value.

Reply:
@s_kanaev. please resolve if this makes sense.